site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -37,24 +36,23 @@
     private final Logger logger = LoggerFactory.getLogger(SiteServiceImpl.class);
 
     /**
-     * @param configurationProvider provide Site Configuration
+     * @param configuration    injected site configuration
      * @param siteStore        Site store interface
      * @param entityValidation Site validation interface
      */
     @Autowired
-    public SiteServiceImpl(SiteConfigurationProvider configurationProvider,
+    public SiteServiceImpl(final SiteConfiguration configuration,
                            final EntityStore<Site, String> siteStore,
-                           ModelEntityValidation<Site> entityValidation) {
-        if (configurationProvider == null) {
-            throw new InitialisationError("site configuration provider cannot be null");
+                           final ModelEntityValidation<Site> entityValidation) {
+        if (configuration == null) {
+            throw new InitialisationError("site configuration cannot be null");
         }
         if (siteStore == null) {
             throw new InitialisationError("site store cannot be null");
         }
         if (entityValidation == null) {
             throw new InitialisationError("entity validation cannot be null");
         }
```

Comment:
IIRC we had a thread to replace internal checks like this with NPE. I know you didn't change this... but it might be a good opportunity if it fits what was discussed in the past. Just a thought...

Reply:
Will replace with calls to Objects.requireNonNull